Implement Initial API Testing Framework for TeamCity

avtikhomirova opened the pull request:
1st iteration

1. Framework Structure: Implemented a comprehensive testing framework covering all essential components.

2. Project Creation Tests: Basic and advanced scenarios including maximum allowed name and ID. Child project creation.
Boundary value analysis.
Negative tests for invalid names, IDs, parent projects, and duplicate projects.

3. Build Configuration Tests: Standard creation process and advanced scenarios with maximum allowed name and ID. Boundary value analysis.
Negative tests for invalid names, IDs, projects, and duplicate build configurations.

avtikhomirova committed:
Implement Initial API Testing Framework for TeamCity
1st iteration

1. Framework Structure: Implemented a comprehensive testing framework covering all essential components.

2. Project Creation Tests:
Basic and advanced scenarios including maximum allowed name and ID.
Child project creation.
Boundary value analysis.
Negative tests for invalid names, IDs, parent projects, and duplicate projects.

3. Build Configuration Tests:
Standard creation process and advanced scenarios with maximum allowed name and ID.
Boundary value analysis.
Negative tests for invalid names, IDs, projects, and duplicate build configurations.

AlexPshe commented on .idea/.gitignore:
Папку .idea не нужно коммитить, она автоматически создается IDE и хранит IDE настройки 🙌

avtikhomirova replied:
Спасибо. добавила в .gitignore

AlexPshe commented on src/test/java/com/example/teamcity/api/BuildConnfigurationTest.java:
```diff
@@ -0,0 +1,158 @@
+                .authSpec(testData.getUser()))
+                .create(buildConfDescription)
+                .then().assertThat().statusCode(HttpStatus.SC_BAD_REQUEST)
+                .body(Matchers.containsString("When creating a build type, non empty name should be provided"));
```
Ошибки можно вынести в отдельный Errors статический класс и сделать константами 🙌

avtikhomirova replied:
Done

AlexPshe commented on src/test/java/com/example/teamcity/api/BuildConnfigurationTest.java:
```diff
@@ -0,0 +1,158 @@
+                .authSpec(testData.getUser()))
+                .create(testData.getBuildType())
+                .then().assertThat().statusCode(HttpStatus.SC_BAD_REQUEST)
+                .body(Matchers.containsString("The build configuration / template ID \""+ testData.getBuildType().getId() +"\" is already used by another configuration or template"));
```
Можно реализовать format для более красивой параметризации строки:
https://stackoverflow.com/questions/17364632/java-generating-strings-with-placeholders

avtikhomirova replied:
Done

AlexPshe reviewed (changes requested):
Привет!

Большое спасибо за качественное выполнение задания!

Я добавила несколько комментов по улучшению - нужно пофиксить их за неделю и будет зачтено ⚡️

avtikhomirova committed:
1. Added .idea to .gitignore
2. Created Abstract class ErrorMessages.java with all errors in constants.
3. Replaced all errors with constants
4. Implemented format for string parametrization in negativeProjectCreationDuplicateId test.

AlexPshe reviewed (approved):
Отлично, все зачтено!
